lib/common/xml.c

```diff
@@ -1111,44 +1136,74 @@ pcmk__xml_escape(const char *text, enum pcmk__xml_escape_type type)
             continue;
         }
 
-        switch (*text) {
-            case '<':
-                g_string_append(copy, "&lt;");
-                break;
-            case '>':
-                g_string_append(copy, "&gt;");
-                break;
-            case '&':
-                g_string_append(copy, "&amp;");
-                break;
-            case '"':
-                if (type == pcmk__xml_escape_attr) {
-                    g_string_append(copy, "&quot;");
-                } else {
-                    g_string_append_c(copy, *text);
-                }
-                break;
-            case '\n':
-                if (type == pcmk__xml_escape_attr) {
-                    g_string_append(copy, "&#x0A;");
-                } else {
-                    g_string_append_c(copy, *text);
+        switch (type) {
+            case pcmk__xml_escape_text:
+                switch (*text) {
+                    case '<':
+                        g_string_append(copy, "&lt;");
```

Comment:
Yeah I can do that in `crmcommon_private.h` unless you prefer `xml_names_internal.h`. It will only be LT, GT, AMP, and QUOT. We could do special case character references (like `"&#x0A;" for newline), but they would be used only within the unit tests.

Reply:
private header is fine